pkg/config/remote/api/http.go

```diff
@@ -256,6 +260,59 @@ func (c *HTTPClient) UpdateAPIKey(apiKey string) {
 	c.headerLock.Unlock()
 }
 
+// NewWebSocket connects to the RC WebSocket backend and returns a new WebSocket
+// connection or a connection / handshake error.
+//
+// The "endpointPath" specifies the resource path to connect to, which is
+// appended to the client baseURL.
+func (c *HTTPClient) NewWebSocket(ctx context.Context, endpointPath string) (*websocket.Conn, error) {
+	// Extract the TLS & Proxy configuration from the HTTP client.
+	transport, ok := c.client.Transport.(*http.Transport)
+	if !ok {
+		return nil, errors.New("failed to extract transport config from HTTP client")
+	}
+
+	dialer := &websocket.Dialer{
+		HandshakeTimeout: 30 * time.Second,
+		TLSClientConfig:  transport.TLSClientConfig,
+		Proxy:            transport.Proxy,
+	}
+
+	// The WebSocket request MUST include the same auth credentials as the plain
+	// HTTP requests.
+	c.headerLock.RLock()
+	headers := maps.Clone(c.header)
+	c.headerLock.RUnlock()
```

Comment:
The WebSocket connections are authenticated in the same way as the RC HTTP requests.